modules/core/arrow-core/src/main/kotlin/arrow/core/extensions/sequenceK.kt

```diff
@@ -71,10 +76,15 @@ interface SequenceKEq<A> : Eq<SequenceK<A>> {
 
   fun EQ(): Eq<A>
 
-  override fun SequenceK<A>.eqv(b: SequenceK<A>): Boolean =
-    zip(b) { aa, bb -> EQ().run { aa.eqv(bb) } }.fold(true) { acc, bool ->
-      acc && bool
-    }
+  /**
+   * This only evaluates up to the first element that differs or to the first element at the index where the other
+   *  sequence is empty
+   */
+  override fun SequenceK<A>.eqv(b: SequenceK<A>): Boolean = object: SequenceKSemialign {}.run {
+    alignWith(this@eqv, b) { ior ->
+      ior.fold({ false }, { false }, { l, r -> EQ().run { l.eqv(r) } })
+    }.firstOption { it.not() }.isEmpty()
+  }
```

Comment:
This is kind of concerning, not the old implementation, but the fact that this never failed a test. (Old one should have failed for `[0, 1] != [0]` for example. Will check that later as well.